packages/opencensus-core/test/test-gauge.ts

```diff
@@ -0,0 +1,407 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import {LabelKey, LabelValue, MetricDescriptorType} from '../src/metrics/export/types';
+import {Gauge} from '../src/metrics/gauges/gauge';
+
+const METRIC_NAME = 'metric-name';
+const METRIC_DESCRIPTION = 'metric-description';
+const UNIT = '1';
+const GAUGE_INT64 = MetricDescriptorType.GAUGE_INT64;
+const GAUGE_DOUBLE = MetricDescriptorType.GAUGE_DOUBLE;
+const LABEL_KEYS: LabelKey[] = [{key: 'code', description: 'desc'}];
+const LABEL_VALUES_200: LabelValue[] = [{value: '200'}];
+const LABEL_VALUES_400: LabelValue[] = [{value: '400'}];
+const LABEL_VALUES_EXRTA: LabelValue[] = [{value: '200'}, {value: '400'}];
+const UNSET_LABEL_VALUE: LabelValue = {
+  value: null
+};
+
+describe('GAUGE_INT64', () => {
+  let instance: Gauge;
+  const expectedMetricDescriptor = {
+    name: METRIC_NAME,
+    description: METRIC_DESCRIPTION,
+    unit: UNIT,
+    type: GAUGE_INT64,
+    labelKeys: LABEL_KEYS
+  };
+
+  beforeEach(() => {
+    instance = new Gauge(
+        METRIC_NAME, METRIC_DESCRIPTION, UNIT, GAUGE_INT64, LABEL_KEYS);
+    process.hrtime = () => [1000, 1e7];
```

Comment:
You redefining `process.hrtime` Its builtin function of global nodejs object, other functionality can use it.
The best practice is to restore the state.

Reply:
Done, here https://github.com/census-instrumentation/opencensus-node/pull/207/commits/164369cf66ad168eaffdbf1422f1b0e0f03c7ba5 
Good catch!